src/Simplex/Messaging/Agent/Store/SQLite/Migrations/M20220322_notifications.hs

```diff
@@ -25,23 +17,22 @@ CREATE TABLE ntf_servers (
   PRIMARY KEY (ntf_host, ntf_port)
 ) WITHOUT ROWID;
 
-CREATE TABLE ntf_subscriptions (
+CREATE TABLE ntf_tokens (
+  platform TEXT NOT NULL, -- apn
+  device_token TEXT NOT NULL,
   ntf_host TEXT NOT NULL,
   ntf_port TEXT NOT NULL,
-  ntf_sub_id BLOB NOT NULL,
-  ntf_sub_status TEXT NOT NULL, -- new, created, active, pending, error_auth
-  ntf_sub_action TEXT, -- if there is an action required on this subscription: create / check / token / delete
-  ntf_sub_action_ts TEXT, -- the earliest time for the action, e.g. checks can be scheduled every X hours
-  ntf_token TEXT NOT NULL, -- or BLOB?
-  smp_host TEXT NOT NULL,
-  smp_port TEXT NOT NULL,
-  smp_ntf_id BLOB NOT NULL,
+  tkn_id BLOB, -- token ID assigned by notifications server
+  tkn_priv_key BLOB NOT NULL, -- private key to sign token commands
+  tkn_pub_key BLOB NOT NULL, -- public key to verify token commands - if it has to be sent to the server again, the same key needs to be sent
+  tkn_dh_secret BLOB, -- DH secret for e2e encryption of notifications
+  tkn_status TEXT NOT NULL,
+  tkn_action TEXT,
+  tkn_action_ts TEXT, -- this is to check token status periodically to know when it was last checked
   created_at TEXT NOT NULL,
-  updated_at TEXT NOT NULL, -- this is to check subscription status periodically to know when it was last checked
-  PRIMARY KEY (ntf_host, ntf_port, ntf_sub_id),
+  updated_at TEXT NOT NULL, -- this is to check token status periodically to know when it was last checked
```

Comment:
a suggestion - maybe it would make sense to have `updated_at` as a service field for any updates and a separate `checked_at` field for the purpose described here?

Reply:
nvm it already seems to be `tkn_action_ts`..